### ISSUE TITLE
[Component] [slider] Error at not integer step

### ISSUE THREAD

**Issue by martonx** [element-plus/element-plus#14802]
Bug Type: **`Component`**

## Environment

- Vue Version: `3.3.8`
- Element Plus Version: `2.4.2`
- Browser / OS: `UserAgent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36 Edg/119.0.0.0`
- Build Tool: `Vite`

### Related Component

- `el-slider`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGRpdiBjbGFzcz1cInNsaWRlci1kZW1vLWJsb2NrXCI+XG4gICAgPGVsLXNsaWRlciB2LW1vZGVsPVwidmFsdWVcIiBzaG93LWlucHV0IG1pbj1cIjAuM1wiIG1heD1cIjIuMVwiIHN0ZXA9XCIwLjFcIiAvPlxuICA8L2Rpdj5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5cbmNvbnN0IHZhbHVlID0gcmVmKDEuMClcbjwvc2NyaXB0PlxuPHN0eWxlIHNjb3BlZD5cbi5zbGlkZXItZGVtby1ibG9jayB7XG4gIGRpc3BsYXk6IGZsZXg7XG4gIGFsaWduLWl0ZW1zOiBjZW50ZXI7XG59XG4uc2xpZGVyLWRlbW8tYmxvY2sgLmVsLXNsaWRlciB7XG4gIG1hcmdpbi10b3A6IDA7XG4gIG1hcmdpbi1sZWZ0OiAxMnB4O1xufVxuPC9zdHlsZT5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsIl9vIjp7fX0=)

### Steps to reproduce

Clicking on minus or plus icons are working fine.
BUT grabbing the slider and move it with mouse not works.

### What is Expected?

when we grab the slider, we can move it.

### What is actually happening?

Q = Number.parseFloat(Q.toFixed(N.value))

SegmentMap-a2349411.js:1  Uncaught (in promise) TypeError: Q.toFixed is not a function

### Additional comments

_(empty)_

**Comment by yj-liuzepeng** [comment]
<img width="1442" alt="image" src="https://github.com/element-plus/element-plus/assets/75007029/2ce88234-b5e2-46f0-b3b7-3520c7e7e257">

 should be set like 👇

<img width="650" alt="image" src="https://github.com/element-plus/element-plus/assets/75007029/1cd3c975-7883-4734-8a5c-62afce904ebd">

**Comment by martonx** [comment]
Agree, but this should be fixed, to work with float numbers. So, if the value is string (float in this case) the slider should work.

**Comment by martonx** [comment]
Agree, but this should be fixed, to work with float numbers. So, if the value is string (float in this case) the slider should work.

**Comment by martonx** [comment]
use-slider-button.mjs

```
const setPosition = async (newPosition) => {
    if (newPosition === null || Number.isNaN(+newPosition))
      return;
    if (newPosition < 0) {
      newPosition = 0;
    } else if (newPosition > 100) {
      newPosition = 100;
    }

    // This could be the fix
    **const maxValue = Number.parseFloat(max.value);
    const minValue = Number.parseFloat(min.value);
    const stepValue = Number.parseFloat(step.value);**
    const lengthPerStep = 100 / ((maxValue - minValue) / stepValue);
    const steps = Math.round(newPosition / lengthPerStep);
    let value = steps * lengthPerStep * (maxValue - minValue) * 0.01 + minValue;
    value = Number.parseFloat(value.toFixed(precision.value));
    if (value !== props.modelValue) {
      emit(UPDATE_MODEL_EVENT, value);
    }
    if (!initData.dragging && props.modelValue !== initData.oldValue) {
      initData.oldValue = props.modelValue;
    }
    await nextTick();
    initData.dragging && displayTooltip();
    tooltip.value.updatePopper();
  };
```

**Comment by Dsaquel** [comment]
I assume you have to pass numbers instead of strings.
